syntax = "proto3";

package adminlogin;

// Request from frontend
message LoginRequest {
    string username = 1;
    string password = 2;
}

// Response from service
message LoginResponse {
    bool success = 1;
    string message = 2;
    int32 admin_id = 3;
}

// New messages for settings
message UpdateAdminRequest {
    int32 admin_id = 1;
    string username = 2; // optional
    string password = 3; // optional
}

message UpdateAdminResponse {
    bool success = 1;
    string message = 2;
}

// Service definition
service AdminLoginService {
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc UpdateAdminInfo(UpdateAdminRequest) returns (UpdateAdminResponse);
}
